test/utils/messaging.ts

```diff
@@ -0,0 +1,28 @@
+import {Signer} from '@ethersproject/abstract-signer';
+import {ethers} from 'hardhat';
+import hre from 'hardhat';
+
+export function applyL1ToL2Alias(l1Address: string): string {
+  const offset = ethers.BigNumber.from(
+      '0x1111000000000000000000000000000000001111',
+  );
+  const l1AddressAsNumber = ethers.BigNumber.from(l1Address);
+
+  const l2AddressAsNumber = l1AddressAsNumber.add(offset);
+
+  const mask = ethers.BigNumber.from(2).pow(160);
+  return l2AddressAsNumber.mod(mask).toHexString();
+}
+
+export async function getL2SignerFromL1(l1Signer: Signer): Promise<Signer> {
```

Comment:
Any particular reason this returns `Promise<Signer>` instead of `Promise<SignerWithAddress>`? Doesn't `ethers.getSigner()` return `Promise<SignerWithAddress>`?

Reply:
Sorry Must have been a typo. Good Catch.